pipelines/tekton/build-container-image-pipeline/build-container-image-pipeline.yaml

```diff
@@ -44,8 +78,6 @@ spec:
         value: registry.redhat.io/openshift-pipelines/pipelines-git-init-rhel8@sha256:1a50511583fc02a27012d17d942e247813404104ddd282d7e26f99765174392c
       - name: subdirectory
         value: /containerfile_repo/
-    runAfter:
-      - kserve-download-model
```

Comment:
IMO, if `git-clone-model-repo` or `kserve-download-model` doesn't run successfully in before `check-model-and-containerfile-exists` then `check-model-and-containerfile-exists` fails and the following task `create-imagestream` doesn't run at all and the pipeline fails. 
The problem with having `runAfter` in both the dependencies is that it doesn't have a conditions field meaning 
 it doesn't support something like runafter: `kserver-download-model` || `git-clone-model-repo`. Let me know if you'd like to discuss this.

Reply:
Why can't you just liste them both
```
    runAfter:
      - kserve-download-model
      - git-clone-model-repo
```
?

The `runAfter` is about ordering -- the fact that one of them will always be skipped so it will be a logical `or` does not matter:
![Screenshot_2023-11-03_16-26-21](https://github.com/opendatahub-io/ai-edge/assets/2536912/4b7326b2-c7ba-4443-9318-8e6538abaf41)

Reply:
I probably mis-understood the runAfter mechanism, that it requires each task below it to successfully run. I'll try this  out.

Reply:
It actually used to be that way so if you knew Tekton before 2021, that's the previous behaviour. It was changed via https://github.com/tektoncd/community/blob/main/teps/0059-skipping-strategies.md, it seems.